clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2322,6 +2396,32 @@ range of [0, 3]. A kernel with max_global_work_dim(0) must be invoked with a
 have arguments of (1, 1, 1).
 If ``intel::max_global_work_dim`` is applied to a function called from a
 device kernel, the attribute is not ignored and it is propagated to the kernel.
+
+.. code-block:: c++
+
+  [[intel::max_global_work_dim(1)]] void foo() {}
+
+  template<int N>
+  [[intel::max_global_work_dim(N)]] void bar() {}
+
+  class Foo {
+  public:
+    [[intel::max_global_work_dim(1)]] void operator()() const {}
+  };
+
+  template <int N>
+  class Functor {
+  public:
+    [[intel::max_global_work_dim(N)]] void operator()() const {}
+  };
+
+  struct TRIFuncObjGood {
```

Comment:
I think it will be clearer to have this as a separate code block with related documentation above it i.e. move following line to above this code block - 

```
if ``intel::max_work_group_size`` or ``cl::reqd_work_group_size`` are applied to the kernel as well - they shall	``cl::reqd_work_group_size`` are applied to the kernel as well - they shall
have arguments of (1, 1, 1). 
```

Reply:
Yes, you are right. I will move this in a separate block to make it clear. Thanks.

Reply:
Done